modules/serverpod_auth/serverpod_auth_server/lib/src/firebase/auth/token_verifier.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:openid_client/openid_client_io.dart';
+import 'package:serverpod_auth_server/src/firebase/errors/firebase_error.dart';
+
+/// JWT verifier
+class TokenVerifier {
+  final String _projectId;
+  final Client? _client;
+
+  /// Creates a new [TokenVerifier] object with a [projectId] and [Client]
+  TokenVerifier({
+    required String projectId,
+    Client? client,
+  })  : _projectId = projectId,
+        _client = client;
+
+  /// Verifies the format and signature of a Firebase Auth JWT token.
+  Future<IdToken> verifyJwt(String jwtToken) async {
+    var client = _client ?? await _getOpenIdClient();
```

Comment:
Question: Instead of mocking away the functionality of the OpenIDClient maybe we could just pass a mock client to it instead? It looks like the constructor takes in a client.